### ISSUE TITLE
Update `TreeView.Item` to specify `aria-selected` on element with `role="treeitem"`

### ISSUE THREAD

**Issue by joshblack** [primer/react#2840]
This issue first came up in: https://github.com/primer/react/pull/2799, specifically with this violation:

```
Elements with the ARIA role "treeitem" must have the following attributes defined: aria-selected  jsx-a11y/role-has-required-aria-props
```

Based on the API document, items with `role="treeitem"` must convey their state via `aria-selected`: https://github.com/github/primer/blob/main/apis/tree-view-api.md#communicating-a-node-has-been-focused-to-assistive-technology

This issue should include updating `TreeView.Item` so that `aria-selected` is specified on the element with `role="treeitem"`
